Add support for markdown checkbox syntax

KyleLH opened the pull request:
Adds another mapping for valid GitHub MarkDown syntax of checkboxes

vitalk commented:
Hi @KyleLH!

A little preamble. When I created this plugin I used a default vim semantic for key bindings (`i`/`I`/`a`/`A`). In the feature I want to make plugin even simpler by providing a single mapping to switch between different item states (item/list item/todo item/etc).

Now your PR:

- The key mapping you're using (`<leader>d`) has no any sense for me, but the plug mapping looks useful. How about set the plug mappings but don't bind them to any key by default? (You always can bind them via your `vimrc`).

- How about a more meaningful name for the plug mapping? The something like `simple-todo-new-list-item` looks better for me.

- Looks like you missed to add mappings into plugin documentation. Please update it as well.

KyleLH commented:
Hey @vitalk! Sounds great, thanks for the feedback! Yeah I wasn't really sure what to assign it to, kind of picked `<leader>d` just because it's easy to type for me, but definitely not including one by default makes sense.

KyleLH commented:
Thanks for the input @vitalk, so I addressed some of your comments, as well as allowing the user to define their own list symbol prefix. Let me know if you see anything off for sure, I'm pretty new to the vimscript world, so there might be some common practices that I'm blatantly unaware of. Cheers!

vitalk commented:
:+1:

KyleLH commented:
Awesome! See anything that looks amiss?

vitalk commented on plugin/simple-todo.vim:
```diff
@@ -36,6 +40,11 @@ endfu " }}}
+" Create a new item with some list prefix symbol
+nnore <silent> <Plug>(simple-todo-new-list-item) "=g:simple_todo_list_symbol.' [ ] '<C-M>pa
+inore <silent> <Plug>(simple-todo-new-list-item) <Esc>"=g:simple_todo_list_symbol.' [ ] '<C-M>pa
+
```
- Please use `<cr>` instead of `<C-M>` (they both are equal, keep `<cr>` for consistency with other lines);
- Remove extra blank line (for consistency as well).

vitalk commented:
Awesome. Thank you!
